cpp/daal/src/algorithms/dbscan/dbscan_partial_result_types.cpp

```diff
@@ -152,26 +152,35 @@ Status DistributedPartialResultStep4::check(const daal::algorithms::Input * inpu
     DataCollectionPtr dcPartitionedPartialOrders = get(partitionedPartialOrders);
 
     DAAL_CHECK_EX(dcPartitionedData.get(), ErrorNullPartialResult, ArgumentName, partitionedDataStr());
-    DAAL_CHECK_EX(dcPartitionedData->size() == nBlocks, ErrorIncorrectDataCollectionSize, ArgumentName, partitionedDataStr());
```

Comment:
You can also add check for `nBlocks == 0` and skip all further checks in such case.